### PR TITLE
[Access] Use Indexed height when checking payer balance

### PR DESCRIPTION
**Pull request by AndriiDiachuk**
In this PR new `IndexedHeight` method was added as part of the `Blocks` interface.

Related PR: https://github.com/onflow/flow-go/pull/6292

### CODE REVIEW

**Comment by peterargue** [comment]
@AndriiDiachuk please fix the merge conflict

File: emulator/blockchain.go [hunk 1]
```diff
@@ -34,6 +34,7 @@ import (
 	"encoding/hex"
 	"errors"
 	"fmt"
+	"github.com/onflow/flow-go/module/metrics"
 	"math"
```
Comment (janezpodhostnik) [inline emulator/blockchain.go:38]: looks like bad linting.

**Review by janezpodhostnik — CHANGES_REQUESTED** [review]
Looks good, just that one linting issue.

**Review by zhangchiqing — APPROVED** [review]

**Comment by peterargue** [comment]
replaced by https://github.com/onflow/flow-emulator/pull/755

**Pull request closed without merging**